pkg/ucp/credentials/aws.go

```diff
@@ -71,6 +71,13 @@ func (p *AWSCredentialProvider) Fetch(ctx context.Context, planeName, name strin
 		default:
 			return nil, errors.New("invalid AWSAccessKeyCredentialProperties")
 		}
+	case *ucpapi.AwsIRSACredentialProperties:
+		switch c := p.Storage.(type) {
+		case *ucpapi.InternalCredentialStorageProperties:
+			storage = c
+		default:
+			return nil, errors.New("invalid AWSAccessKeyCredentialProperties")
+		}
```

Comment:
is it possible to  combine both these?
eg. case *ucpapi.AwsAccessKeyCredentialProperties,  *ucpapi.AwsIRSACredentialProperties

Reply:
nice one. But it does not like it when I tried, since in the next line we  have `switch c := p.Storage.(type)` and looks since now p can correspond to either of 2 values, it gives an error.

Reply:
let's pull common code into function and each switch case calls the function, eg:
func getStorageProperties(p any) (*InternalCredentialStorageProperties, error) {
	switch c := p.(type) {
	case *InternalCredentialStorageProperties:
		return c, nil
	default:
		return nil, errors.New("invalid AWS credential storage properties")
	}
}